### PR TITLE
feat(network-diagnostics): Install bones network debug window

### PR DESCRIPTION
**Pull request by MaxCWhitehead**
Install the bones network debug egui plugin / add a button to open it from debug window.

Companion PR to https://github.com/fishfolk/bones/pull/345

### CODE REVIEW

**Comment by zicklag** [comment]
I merged the bones PR so you should be able to run `cargo update` to make CI pass for this now.

File: src/core/physics/collisions.rs [hunk 1]
```diff
@@ -301,7 +301,7 @@ impl<'a> CollisionWorld<'a> {
             if !self.entities.is_alive(entity) {
                 // Remove any collisions with the killed entity from the collision cache.
                 let mut collisions = collision_cache.collisions.borrow_mut();
-                let colliding_with = collisions.remove(&entity);
+                let colliding_with = collisions.swap_remove(&entity);
```
Comment (zicklag) [inline src/core/physics/collisions.rs:304]: What is the effect of this change?

Comment (MaxCWhitehead) [reply src/core/physics/collisions.rs:304]: Ah I should've commented on that - remove was deprecated in index-map crate due to ambiguity. Remove already did a `swap_remove`, they deprecated and require calling `swap_remove` or `shift_remove`, to clarify if user intends to remove and preserve ordering, or swap with last element for performance while modifying ordering.

Behavior has not changed, fixes deprecation warning. See context:

https://github.com/indexmap-rs/indexmap/blob/184fe4bfcba20e21242c924220170b98be157d45/src/map.rs#L645-L653

**Review by zicklag — APPROVED** [review]

**Comment by MaxCWhitehead** [comment]
Merged main to resolve cargo.lock conflict

**Comment by zicklag** [comment]
I think there might be a security warning that is going to block merging. Hopefully we can just run `cargo update` to fix it if that's the case.

**Comment by MaxCWhitehead** [comment]
> I think there might be a security warning that is going to block merging. Hopefully we can just run `cargo update` to fix it if that's the case.

I just ran cargo update in the merge from main - so it seems these are still around. I can take a swing at resolving some of these cargo deny errors if you would like.

**Comment by zicklag** [comment]
Oh, I've already got the advisories set as non-required for merge, so it looks like all we need to do is add `Unicode-3.0` to our allowed licenses list.

**Comment by MaxCWhitehead** [comment]
> Oh, I've already got the advisories set as non-required for merge, so it looks like all we need to do is add `Unicode-3.0` to our allowed licenses list.

PR opened: https://github.com/fishfolk/jumpy/pull/923

**Pull request merged**